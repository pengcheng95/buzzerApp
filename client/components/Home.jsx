import React from 'react';
import { Layout, Row, Col, Input, Icon, Button, notification } from 'antd';
import { io } from '../sockets/config.jsx';

const { Header, Content, Footer } = Layout;

class Home extends React.Component {

  state = {
    createRoom: '',
    joinRoom: ''
  }

  componentDidMount() {
    io.roomTaken(() => {
      notification['warning']({
        message: 'Room Name Taken',
        description: 'Input another name to create the room',
        style: {
          width: 300,
          marginLeft: 100
        },
      })
    })

    io.roomCreated((data) => {
      this.props.history.push(`room/${data}`);
    })

    io.joinedRoom((data) => {
      this.props.history.push(`room/${data}`);
    })

    io.roomNoExist(() => {
      notification['warning']({
        message: 'Room Name Does Not Exist',
        description: 'Check to see if you inputed the correct room name or create room.',
        style: {
          width: 300,
          marginLeft: 100
        },
      })
    })
  }

  handleChange = (e) => {
    let name = e.target.name;
    let val = e.target.value;
    this.setState({
      [name]: val
    }, () => {
      console.log(this.state)
    })
  }

  createRoom = (e) => {
    io.createRoom(this.state.createRoom);
  }

  joinRoom = (e) => {
    io.joinRoom(this.state.joinRoom);
  }


	render() {
    let inputPrefixStyle = { color: 'rgba(0,0,0,.25)' };
    let inputStyle = {width: '90%', clear: 'right'};
    let buttonStyle = {marginTop: '30px', marginBottom: '20px'};
    let layoutStyle = {
      background: '#ece9e6', /* fallback for old browsers */
      background: '-webkit-linear-gradient(to bottom, #ece9e6, #ffffff)', /* Chrome 10-25, Safari 5.1-6 */
      background: 'linear-gradient(to bottom, #ece9e6, #ffffff)'
      // background: '#c9d6ff', /* fallback for old browsers */
      // background: '-webkit-linear-gradient(to bottom, #c9d6ff, #e2e2e2)', /* Chrome 10-25, Safari 5.1-6 */
      // background: 'linear-gradient(to bottom, #c9d6ff, #e2e2e2)' /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

		return (
		  <Layout className="layout" style={layoutStyle}>
        <Content style={{ padding: '20px' }}>
          <div style={{ padding: 12, minHeight: '70vh' }}>
            <Row>
              <Col span={12} offset={6} style={{ textAlign: 'center', paddingTop: '30px' }}>
                <h3> Create a Room </h3>
                <Input 
                  prefix={<Icon type="laptop" style={inputPrefixStyle}/>}
                  style={inputStyle}
                  name="createRoom"
                  value={this.state.createRoom}
                  onChange={this.handleChange}
                />
                <Button type="primary" style={buttonStyle} onClick={this.createRoom}> Create </Button>

                <h4 style={{marginTop: '5px', marginBottom: '15px'}}> - OR - </h4>

                <h3> Join a Room </h3>
                <Input 
                  prefix={<Icon type="laptop" style={inputPrefixStyle}/>}
                  style={inputStyle}
                  name="joinRoom"
                  value={this.state.joinRoom}
                  onChange={this.handleChange}
                />
                <Button type="primary" style={buttonStyle} onClick={this.joinRoom}> Join </Button>
              </Col>
            </Row>
          </div>
        </Content>
      </Layout>
    )
	}
}

export default Home;