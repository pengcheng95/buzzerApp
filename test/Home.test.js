import React from 'react';
import Home from '../client/components/Home.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Home/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});