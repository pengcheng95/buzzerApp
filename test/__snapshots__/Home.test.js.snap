// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="layout ant-layout"
  style={
    Object {
      "background": "linear-gradient(to bottom, #ece9e6, #ffffff)",
    }
  }
>
  <div
    className="ant-layout-content"
    style={
      Object {
        "padding": "20px",
      }
    }
  >
    <div
      style={
        Object {
          "minHeight": "70vh",
          "padding": 12,
        }
      }
    >
      <div
        className="ant-row"
        style={undefined}
      >
        <div
          className="ant-col-12 ant-col-offset-6"
          style={
            Object {
              "paddingTop": "30px",
              "textAlign": "center",
            }
          }
        >
          <h3>
             Create a Room 
          </h3>
          <span
            className="ant-input-affix-wrapper"
            style={
              Object {
                "clear": "right",
                "width": "90%",
              }
            }
          >
            <span
              className="ant-input-prefix"
            >
              <i
                className="anticon anticon-laptop"
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
              />
            </span>
            <input
              className="ant-input"
              disabled={false}
              name="createRoom"
              onChange={[Function]}
              onKeyDown={[Function]}
              style={null}
              type="text"
              value=""
            />
          </span>
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": "20px",
                "marginTop": "30px",
              }
            }
            type="button"
          >
            <span>
               Create 
            </span>
          </button>
          <h4
            style={
              Object {
                "marginBottom": "15px",
                "marginTop": "5px",
              }
            }
          >
             - OR - 
          </h4>
          <h3>
             Join a Room 
          </h3>
          <span
            className="ant-input-affix-wrapper"
            style={
              Object {
                "clear": "right",
                "width": "90%",
              }
            }
          >
            <span
              className="ant-input-prefix"
            >
              <i
                className="anticon anticon-laptop"
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
              />
            </span>
            <input
              className="ant-input"
              disabled={false}
              name="joinRoom"
              onChange={[Function]}
              onKeyDown={[Function]}
              style={null}
              type="text"
              value=""
            />
          </span>
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": "20px",
                "marginTop": "30px",
              }
            }
            type="button"
          >
            <span>
               Join 
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
